src/lib.rs

```diff
@@ -1010,6 +1012,42 @@ pub trait Itertools : Iterator {
         None
     }
 
+    /// Find the positions of each element satisfying a predicate.
+    ///
+    /// Equivalent to `iter.filter(|(_,v)| pred(v)).map(|(i,_)| i)`.
```

Comment:
missing enumerate in the equivalency.

I'd make the doc even more clear by saying something like "yields the index of each element matching the predicate, counted from the start of the iterator".

Reply:
The thought is that we need to remind the user somehow that we count indices as we see them in the iterator.  If they give us `slice[3..].iter().positions()` we are not counting indices in the original slice etc.